sky/provision/vsphere/instance.py

```diff
@@ -0,0 +1,604 @@
+"""Vsphere instance provisioning."""
+import json
+import os
+from typing import Any, Dict, List, Optional
+
+from com.vmware.vcenter_client import VM
+import pandas as pd
+from pyVmomi import vim  # type: ignore
+
+from sky import exceptions
+from sky import sky_logging
+from sky import status_lib
+from sky.clouds.service_catalog.common import get_catalog_path
+from sky.provision import common
+from sky.provision.vsphere import vsphere_utils
+from sky.provision.vsphere.common.custom_script import CUSTOMIZED_SCRIPT
+from sky.provision.vsphere.common.metadata_utils import Metadata
```

Comment:
style nit: we normally import modules only, not the classes directly. Reference: https://google.github.io/styleguide/pyguide.html#22-imports

Reply:
fixed